text/0019-clipboard-rearchitecture.md

```diff
@@ -0,0 +1,364 @@
+
+Electron provided a [`clipboard`](https://www.electronjs.org/docs/latest/api/clipboard) API to its users preceding the specification of the [Clipboard API by the W3C](https://w3c.github.io/clipboard-apis/#clipboard-interface). As a result, we locked into an API surface and created an API contract with our users that became progressively harder to alter without breaking said contract while the web drove a more standardized approach. On top of this confusion, the [`clipboard`](https://www.electronjs.org/docs/latest/api/clipboard) module provided by Electron is a dual-process module, meaning that it exists in parallel to the more robust API provided by Chromium in the renderer process via `navigator.clipboard` global. 
+
+At this point, it becomes clear that it is in users' and maintainers' best interest to reconsider how we approach the `clipboard` module - breaking changes might be difficult in the short term but provide more consistency and functionality as we look to the future of the module.
+
+## Guide-level explanation
+
+### API Design
+
+The largest consideration for changes to the `clipboard` module is to what degree we should choose to align with the W3C specification. As we consider this, we should consider what needs or use cases might exist for Electron uses which necessarily wouldn't exist for Chrome users, and how we would plan to account for that in our new design.
+ 
+#### Preferred Approach/Solution
+
+This design recommends alignment with W3C where possible, with some additional APIs exposed to handle desktop use cases which aren’t considered by the specification.  Given this approach, we will preserve some of the existing APIs, remove some of the existing APIs, and, modify the rest of the APIs to be in compliance with the W3C spec as much as possible.
+
+**APIs to add**
+
+### Event: 'clipboardchange'
+
+Returns:
+
+* `event` Event
+  * `types` string[] An array of MIME types that are available in the clipboard.
+
+The clipboardchange event fires whenever the contents of the system clipboard are changed.
+
+On Linux, instead of passing a type of `selection` to access the selection clipboard, the following
+APIS will be added to work with the selection clipboard:
+* `clipboard.selection.clear()` _Linux_
+  * Clears the selection clipboard.  Replaces `clipboard.clear('selection')`.
+* `clipboard.selection.has(format)` _Linux_
+  * Check if a particular format is available on the selection clipboard.  Replaces `clipboard.has(format,'selection)`.
+* `clipboard.selection.read()` _Linux_
+  * Reads from the selection clipboard following the [Web API clipboard.read](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/read) spec.  Replaces `clipboard.read('selection')` and `clipboard.readBuffer('selection)`.
```

Comment:
This function has an optional parameter to request reading only particular formats https://w3c.github.io/clipboard-apis/#dom-clipboard-read. Is it left out on purpose here ?

Reply:
My bad! Thanks for clearing that up.